
# CX Radar: Mapeando Experiências Urbanas  
## Além do NPS: Inteligência de Reclamações
---

### Introdução

Este projeto propõe uma abordagem analítica inovadora para entender a experiência do cliente em serviços urbanos, utilizando dados públicos de corridas de táxi como analogia. Através de métricas como tempo de viagem, tarifa, distância e índices derivados, buscamos identificar padrões que possam representar atritos na jornada do usuário.

Mais do que medir satisfação, o projeto avança para uma proposta de classificação entre filiais (simuladas por zonas de origem), considerando não apenas indicadores operacionais, mas também o contexto das reclamações registradas. A ideia é diferenciar falhas humanas de falhas sistêmicas, permitindo uma avaliação mais justa e estratégica da performance local.

Este estudo é um convite para repensar o papel da análise de dados na gestão da experiência — indo além do NPS, além dos dashboards genéricos, e mergulhando em uma leitura mais profunda e contextualizada do que realmente afeta o cliente.

---

### 🎯 1. Objetivo

Este notebook tem como objetivo analisar dados de corridas de táxi na cidade de Nova York para identificar padrões que possam impactar a experiência do cliente. Através de métricas como tempo de viagem, tarifa e distância, buscamos compreender como esses fatores influenciam a percepção de valor e satisfação.

---

### 🌍 2. Contexto

A experiência do cliente (CX) é um dos principais pilares de retenção e fidelização em serviços urbanos. No setor de mobilidade, fatores como tempo de espera, custo da corrida e localização influenciam diretamente a jornada do usuário. Este estudo utiliza dados públicos de corridas de táxi como analogia para mapear pontos de atrito e oportunidades de melhoria.

---

### 📦 3. Dataset
Utilizamos o conjunto de dados `samples.nyctaxi.trips`, disponível na plataforma Databricks Community Edition. O dataset contém registros de corridas de táxi em Nova York, incluindo informações como horário de embarque e desembarque, distância percorrida, valor da tarifa e localização de origem/destino.

---

### 🧪 4. Metodologia
As análises foram realizadas por meio de consultas SQL diretamente no notebook. As métricas foram agrupadas e visualizadas em gráficos interativos no dashboard `CX_Taxi_Insights`. A abordagem foi exploratória, com foco em identificar padrões que possam representar boas ou más experiências para o cliente. Também criamos métricas derivadas, como o índice de frustração, para enriquecer a análise.

---

### 🔍 5. Primeira Fase: Exploração de Métricas Individuais

- Tempo médio por hora
- Tarifa média por distância
- Índice de frustração
- Visualizações iniciais

📌 *Conclusão:* Identificamos padrões que sugerem atritos operacionais em horários e zonas específicas.

---

### 🧩 6. Segunda Fase: Classificação Simulada por Zona

- Racional: tratar zonas como filiais urbanas
- Indicadores escolhidos: tempo, tarifa, frustração, proporção de corridas caras
- Normalização e cálculo da nota final
- Visual: ranking por zona

📌 *Conclusão:* Algumas zonas apresentam notas abaixo de 6, indicando risco de insatisfação.

---

### 🚨 7. Achados Críticos

- Zonas com frustração elevada
- Tarifas desproporcionais
- Ineficiências operacionais

📌 *Recomendações:* Monitorar zonas críticas, revisar rotas e precificação, criar alertas preditivos.

---

### 📊 8. Simulação de Dashboard Corporativo

- Indicador principal (gauge ou rosca)
- Aba “Zonas em Alerta”
- Tabela com filtros e insights
- Slide executivo com resumo e recomendações

📌 *Objetivo:* Mostrar como a análise pode ser integrada a um dashboard usado por líderes e gerentes.

---

### 🧠 9. Reflexão Final

- Como análises exploratórias podem evoluir para modelos de decisão
- A importância de contextualizar dados operacionais com visão de cliente
- Potencial de replicação em ambientes reais (como Bemol)



###🕒 1. Tempo médio de viagem por hora

🔍 Pergunta
Quais horários do dia apresentam maior tempo médio de viagem, e como isso pode impactar a experiência do cliente?

🧠 Análise
Agrupamos as corridas por hora de embarque (tpep_pickup_datetime) e calculamos o tempo médio de viagem em minutos. Essa métrica revela os períodos do dia em que os deslocamentos são mais lentos, possivelmente devido a fatores como trânsito, demanda elevada ou disponibilidade de motoristas.

Os horários com maior tempo médio podem indicar janelas de atrito operacional, onde o cliente enfrenta uma jornada mais longa — o que pode gerar frustração, especialmente em corridas curtas ou com tarifa elevada. Esses insights são fundamentais para entender a dinâmica urbana e antecipar momentos críticos na experiência.

In [0]:
SELECT 
  HOUR(tpep_pickup_datetime) AS hora,
  AVG(DATEDIFF(MINUTE, tpep_pickup_datetime, tpep_dropoff_datetime)) AS tempo_medio
FROM samples.nyctaxi.trips
GROUP BY hora
ORDER BY hora;


hora tempo_medio 0 14.163758389261744 1 13.93193717277487 2 20.49438202247191 3 20.324675324675326 4 17.508 5 10.377272727272727 6 10.204210526315789 7 12.292134831460674 8 13.315942028985507 9 13.799808429118775 10 12.7139896373057 11 15.770750988142293 12 16.807547169811322 13 14.459780621572213 14 15.122854561878952 15 17.091296928327644 16 13.103483606557377 17 16.49705139005897 18 15.173883161512027 19 14.139700641482538 20 13.241987179487179 21 11.869096934548468 22 16.710217755443885 23 14.247907949790795

Databricks visualization. Run in Databricks to view.

## 2. Tarifa média por faixa de distância
### 🔍Pergunta
Clientes que fazem corridas curtas pagam mais proporcionalmente?

### 🧠 Análise
Agrupamos as corridas por distância arredondada e calculamos a tarifa média. Isso revela se há distorções na percepção de valor.

In [0]:
SELECT 
  ROUND(trip_distance, 0) AS faixa_distancia,
  AVG(fare_amount) AS tarifa_media
FROM samples.nyctaxi.trips
WHERE trip_distance > 0
GROUP BY faixa_distancia
ORDER BY faixa_distancia;


faixa_distancia tarifa_media 0.0 4.393352144469526 1.0 6.568931702814951 2.0 10.024295432458699 3.0 13.313186813186814 4.0 16.147385103011093 5.0 19.13295719844358 6.0 21.47265625 7.0 24.135135135135137 8.0 26.25 9.0 29.175276752767527 10.0 31.590733590733592 11.0 33.90710382513661 12.0 37.43307086614173 13.0 41.324561403508774 14.0 42.785714285714285 15.0 45.225 16.0 50.29054054054054 17.0 51.55660869565217 18.0 52.76495726495727 19.0 53.06547619047619 20.0 53.72115384615385 21.0 64.046875 22.0 53.6 23.0 59.785714285714285 24.0 72.0 25.0 73.16666666666667 26.0 48.166666666666664 31.0 95.0

Databricks visualization. Run in Databricks to view.

## 3. Zonas com maior tarifa média
### 🔍 Pergunta
Quais regiões têm tarifas mais altas?

### 🧠 Análise
Listamos as 5 zonas de origem com maior tarifa média. Essas áreas podem gerar insatisfação ou exigir atenção estratégica.

In [0]:
SELECT 
  pickup_zip,
  COUNT(*) AS total_corridas,
  AVG(fare_amount) AS tarifa_media
FROM samples.nyctaxi.trips
GROUP BY pickup_zip
ORDER BY tarifa_media DESC
LIMIT 5;


pickup_zip total_corridas tarifa_media 8876 1 260.0 7974 1 188.0 7310 1 105.0 7114 1 105.0 7311 1 60.0

Databricks visualization. Run in Databricks to view.

##  4. Tempo total da corrida
### 🔍 Pergunta
Qual é o tempo médio total das corridas?

### 🧠 Análise
Essa métrica mostra a duração média das corridas, útil para entender a agilidade do serviço.

In [0]:
SELECT 
  AVG(DATEDIFF(MINUTE, tpep_pickup_datetime, tpep_dropoff_datetime)) AS tempo_medio_corrida
FROM samples.nyctaxi.trips;


tempo_medio_corrida 14.62725697610797

Databricks visualization. Run in Databricks to view.

## 5. Tarifa por minuto
### 🔍 Pergunta
Quanto o cliente paga por minuto de corrida?

### 🧠 Análise
Ajuda a entender a percepção de valor — especialmente em corridas curtas.

In [0]:
SELECT 
  ROUND(AVG(fare_amount / DATEDIFF(MINUTE, tpep_pickup_datetime, tpep_dropoff_datetime)), 2) AS tarifa_por_minuto
FROM samples.nyctaxi.trips
WHERE DATEDIFF(MINUTE, tpep_pickup_datetime, tpep_dropoff_datetime) > 0;


tarifa_por_minuto 1.1

Databricks visualization. Run in Databricks to view.

## 6. Corridas com tarifa acima da média
### 🔍 Pergunta
Quais corridas tiveram tarifas muito acima da média?

### 🧠 Análise
Identifica outliers que podem gerar reclamações ou insatisfação.

In [0]:
SELECT 
  fare_amount,
  trip_distance,
  tpep_pickup_datetime
FROM samples.nyctaxi.trips
WHERE fare_amount > (SELECT AVG(fare_amount) FROM samples.nyctaxi.trips)
ORDER BY fare_amount DESC
LIMIT 10;


fare_amount trip_distance tpep_pickup_datetime 275.0 20.85 2016-02-12T20:55:19.000Z 260.0 0.0 2016-02-29T12:16:16.000Z 188.0 0.0 2016-01-30T22:28:42.000Z 130.0 25.46 2016-02-17T22:23:14.000Z 115.0 21.3 2016-01-28T17:36:17.000Z 105.0 0.0 2016-01-04T18:58:23.000Z 105.0 0.0 2016-01-16T18:09:15.000Z 95.0 30.6 2016-02-22T21:17:27.000Z 95.0 12.49 2016-02-24T22:19:55.000Z 95.0 5.2 2016-01-04T09:19:53.000Z

Databricks visualization. Run in Databricks to view.

## 7. Tempo médio por dia da semana
### 🔍 Pergunta
A experiência varia conforme o dia da semana?

### 🧠 Análise
Mostra se há dias com maior tempo de viagem → possível impacto na satisfação.

In [0]:
SELECT 
  DAYOFWEEK(tpep_pickup_datetime) AS dia_semana,
  AVG(DATEDIFF(MINUTE, tpep_pickup_datetime, tpep_dropoff_datetime)) AS tempo_medio
FROM samples.nyctaxi.trips
GROUP BY dia_semana
ORDER BY dia_semana;


dia_semana tempo_medio 1 11.919381375452451 2 12.419582772543741 3 14.659841954022989 4 13.969121140142517 5 17.183508989460634 6 16.330900926185798 7 15.31020288150544

Databricks visualization. Run in Databricks to view.

## 8. Índice de frustração
### 🔍 Pergunta
Quais corridas têm maior potencial de frustração?

### 🧠 Análise
Criamos uma métrica fictícia: tempo * tarifa / distância. Quanto maior, mais provável que o cliente esteja insatisfeito.

### ✨ Insight
As corridas com maior índice de frustração revelam situações onde o cliente pode ter sentido que o serviço não valeu o custo ou o tempo investido. Essa análise pode ser aplicada em contextos de mobilidade, logística ou atendimento, ajudando a antecipar insatisfações e redesenhar jornadas.

In [0]:
SELECT 
  trip_distance,
  fare_amount,
  DATEDIFF(MINUTE, tpep_pickup_datetime, tpep_dropoff_datetime) AS tempo,
  ROUND((DATEDIFF(MINUTE, tpep_pickup_datetime, tpep_dropoff_datetime) * fare_amount) / trip_distance, 2) AS indice_frustracao
FROM samples.nyctaxi.trips
WHERE trip_distance > 0
ORDER BY indice_frustracao DESC
LIMIT 10;


trip_distance fare_amount tempo indice_frustracao 1.0 10.0 1414 14140.0 0.5 4.5 1435 12915.0 0.62 5.5 1437 12747.58 0.45 6.0 895 11933.33 0.74 6.0 1435 11635.14 0.73 5.5 1430 10773.97 1.09 7.5 1431 9846.33 1.37 9.0 1436 9433.58 2.05 13.5 1414 9311.71 2.1 13.5 1404 9025.71

Databricks visualization. Run in Databricks to view.

## 9. Corridas curtas com tarifa alta

### 🔍 Pergunta  
Clientes que fizeram corridas muito curtas pagaram tarifas elevadas?

### 🧠 Análise  
Esta consulta busca identificar corridas com menos de 1 milha de distância, mas com tarifas superiores a 10 dólares. Embora curtas, essas corridas podem gerar uma percepção negativa de valor, especialmente se o cliente sentir que pagou caro por um trajeto pequeno. Situações como trânsito intenso, tempo de espera ou tarifas mínimas elevadas podem contribuir para esse cenário.

Essa análise é relevante para entender como a combinação de baixa distância e alto custo pode afetar a experiência do cliente, principalmente em contextos urbanos onde deslocamentos curtos são comuns. Identificar esses casos ajuda a antecipar insatisfações e propor ajustes na precificação ou comunicação.

In [0]:
SELECT 
  trip_distance,
  fare_amount
FROM samples.nyctaxi.trips
WHERE trip_distance < 1 AND fare_amount > 10
ORDER BY fare_amount DESC;


trip_distance fare_amount 0.0 260.0 0.0 188.0 0.0 105.0 0.0 105.0 0.92 60.0 0.18 55.0 0.2 52.0 0.0 52.0 0.0 52.0 0.0 52.0 0.03 52.0 0.0 52.0 0.0 52.0 0.0 52.0 0.3 52.0 0.0 52.0 0.0 52.0 0.12 52.0 0.0 52.0 0.1 52.0 0.0 52.0 0.0 52.0 0.0 52.0 0.0 52.0 0.0 52.0 0.0 52.0 0.0 50.0 0.0 40.0 0.0 36.34 0.0 27.5 0.83 25.5 0.58 22.5 0.3 20.5 0.0 20.0 0.7 20.0 0.0 20.0 0.4 17.0 0.9 17.0 0.97 15.5 0.65 14.5 0.0 14.0 0.8 13.5 0.0 13.3 0.98 13.0 0.7 13.0 0.89 13.0 0.62 13.0 0.93 13.0 0.86 12.5 0.69 12.5 0.8 12.0 0.76 12.0 0.4 12.0 0.9 12.0 0.7 12.0 0.99 12.0 0.6 12.0 0.7 12.0 0.9 12.0 0.6 11.5 0.99 11.5 0.76 11.5 0.78 11.5 0.84 11.5 0.5 11.5 0.9 11.5 0.9 11.5 0.7 11.0 0.85 11.0 0.6 11.0 0.9 11.0 0.76 11.0 0.9 11.0 0.8 11.0 0.7 11.0 0.7 11.0 0.68 11.0 0.0 11.0 0.93 11.0 0.82 11.0 0.83 10.5 0.9 10.5 0.86 10.5 0.72 10.5 0.98 10.5 0.83 10.5 0.85 10.5 0.7 10.5 0.8 10.5 0.61 10.5 0.81 10.5 0.9 10.5 0.85 10.5 0.0 10.3

Databricks visualization. Run in Databricks to view.

## Classificação Simulada por Zonas de Origem

Nesta seção, tratamos cada zona de embarque (`pickup_zip`) como uma “filial” urbana. A proposta é calcular uma nota de experiência por zona com base em indicadores operacionais que podem refletir atritos na jornada do cliente.

A nota será composta por:

- Tempo médio de viagem
- Tarifa média
- Índice de frustração médio
- Proporção de corridas com tarifa acima da média

Cada indicador será normalizado e ponderado para gerar uma nota final entre 0 e 10.


### 🧭 Tempo médio de viagem por zona
🔍 Pergunta
Quais zonas urbanas apresentam os maiores tempos médios de viagem?

🧠 Análise
Agrupamos as corridas por zona de embarque (pickup_zip) e calculamos o tempo médio de deslocamento em minutos. Essa métrica revela quais regiões têm viagens mais longas, o que pode indicar problemas como congestionamento, rotas ineficientes ou baixa disponibilidade de motoristas.

Zonas com tempo médio elevado tendem a gerar maior frustração para o cliente, especialmente quando combinadas com tarifas altas ou distâncias curtas. Esse indicador é um dos pilares da nota final simulada de experiência.

📌 Exemplo de achado: A zona 11430 apresenta tempo médio de 73 minutos — mais que o dobro da média geral — sugerindo um ponto crítico na operação.

In [0]:
-- Tempo médio de viagem por zona
SELECT 
  pickup_zip,
  ROUND(AVG(DATEDIFF(MINUTE, tpep_pickup_datetime, tpep_dropoff_datetime)), 2) AS tempo_medio
FROM samples.nyctaxi.trips
WHERE pickup_zip IS NOT NULL
GROUP BY pickup_zip
ORDER BY tempo_medio DESC;


pickup_zip tempo_medio 11430 73.25 11420 48.0 11422 41.47 7087 38.0 11375 30.33 11371 30.05 11218 30.0 11235 29.5 11436 28.91 11370 28.12 11221 26.5 11435 24.0 11417 23.5 11424 23.5 7737 23.0 10007 22.72 10111 20.68 11226 20.25 10470 20.0 10305 20.0 11207 19.5 10103 18.5 10119 18.43 10154 18.12 10014 17.74 10025 16.88 11230 16.33 10032 16.27 10005 16.08 11423 16.0 11215 16.0 10271 15.94 10065 15.69 11225 15.6 11209 15.5 11373 15.5 10020 15.47 10006 15.45 10280 15.33 11377 15.29 11222 15.19 10038 15.15 10018 15.13 11205 14.95 11216 14.88 10199 14.75 10282 14.66 10011 14.49 10002 14.36 11224 14.33 10278 14.23 10001 14.01 10013 13.95 10017 13.53 10022 13.42 10171 13.42 10009 13.38 11104 13.07 7002 13.0 10037 12.96 10112 12.74 10012 12.72 11211 12.65 11201 12.56 11217 12.3 11109 12.3 10167 12.26 11102 12.06 11208 12.0 11368 12.0 11210 12.0 10452 12.0 11237 12.0 11101 11.95 11238 11.94 10003 11.93 10023 11.88 10016 11.83 10028 11.8 10034 11.78 10110 11.73 10019 11.57 10044 11.44 10152 11.28 11231 11.2 10029 11.15 10021 11.02 11233 11.0 10026 10.94 10115 10.93 10153 10.81 11372 10.53 11206 10.52 11105 10.46 10027 10.43 10069 10.4 10030 10.28 11103 10.19 10162 10.01 10474 10.0 11356 10.0 10035 9.95 10031 9.63 11355 9.0 10460 9.0 11106 8.9 10451 8.5 11369 7.8 10039 7.6 11378 7.0 11223 7.0 10024 5.0 7718 4.0 11220 4.0 11418 3.0 10454 3.0 11212 2.0 10468 1.5 7974 1.0 7030 0.0 7310 0.0 10463 0.0 11003 0.0 7311 0.0 7114 0.0 7086 0.0 10502 0.0 8876 0.0

### 💰 Tarifa média por zona
🔍 Pergunta
Quais zonas urbanas apresentam as tarifas médias mais altas, e o que isso pode indicar sobre a experiência do cliente?

🧠 Análise
Agrupamos as corridas por zona de embarque (pickup_zip) e calculamos a média das tarifas cobradas. Essa métrica revela quais regiões têm corridas mais caras em média, o que pode estar relacionado à distância, demanda, ou até práticas de precificação local.

Zonas com tarifa média elevada podem gerar percepção de custo excessivo, especialmente se combinadas com tempo de viagem alto ou baixa eficiência. Esse indicador é essencial para entender o valor percebido pelo cliente e compõe 30% da nota final simulada de experiência.

📌 Exemplo de achado: A zona 8876 apresenta tarifa média de $260 — muito acima da média geral — sugerindo um ponto de atenção na precificação.

In [0]:
-- Próximo passo: Tarifa média por zona
SELECT 
  pickup_zip,
  ROUND(AVG(fare_amount), 2) AS tarifa_media
FROM samples.nyctaxi.trips
WHERE pickup_zip IS NOT NULL
GROUP BY pickup_zip
ORDER BY tarifa_media DESC;


pickup_zip tarifa_media 8876 260.0 7974 188.0 7114 105.0 7310 105.0 7311 60.0 11430 52.0 11420 52.0 11368 52.0 11422 44.78 11436 42.32 7086 40.0 7030 40.0 11424 34.25 11235 33.25 11375 33.0 7087 31.0 11371 30.64 10470 28.0 11378 27.25 11370 25.31 11417 25.0 11369 24.2 11221 21.75 11230 20.83 11207 20.5 11224 20.33 11218 20.0 11209 19.25 11208 19.0 11226 18.75 11435 18.17 11109 17.8 10034 16.67 11373 16.63 10032 16.63 10280 16.33 10038 15.9 10005 15.87 10271 15.8 11216 15.63 7737 15.5 10006 15.22 11356 15.0 11101 14.7 11225 14.6 10282 14.46 11222 14.24 11205 13.95 10452 13.88 10007 13.86 10013 13.84 11210 13.75 11215 13.73 10305 13.5 11201 13.46 10037 13.4 10035 13.27 11377 13.25 10278 12.99 10103 12.61 10112 12.59 11237 12.4 11372 12.37 11105 12.31 11211 12.3 11217 12.25 11104 12.24 10002 12.23 10171 12.04 11231 12.03 10154 11.9 10115 11.89 11102 11.88 10020 11.7 10025 11.68 10167 11.58 11423 11.5 11238 11.44 10018 11.4 10012 11.35 10017 11.31 10009 11.16 11103 11.03 11206 11.02 11355 11.0 10003 10.98 10016 10.96 10019 10.95 10014 10.92 10027 10.92 10044 10.91 10011 10.91 10119 10.91 10110 10.9 10029 10.79 10030 10.75 10199 10.74 10111 10.69 10001 10.62 7002 10.5 10026 10.4 10031 10.29 11106 10.24 10028 10.21 10021 10.21 10152 10.13 10162 10.08 10023 10.04 10022 9.84 10065 9.81 10069 9.78 10153 9.77 11233 9.5 10460 9.33 10451 8.63 10474 8.5 10039 7.7 11223 7.5 10024 6.5 11220 6.0 7718 5.0 11418 5.0 10454 5.0 10468 4.0 11212 4.0 10463 0.01 10502 0.0 11003 0.0

### 😣 Índice de frustração médio por zona
🔍 Pergunta
Quais zonas urbanas apresentam maior frustração média, considerando tempo, tarifa e distância?

🧠 Análise
Criamos uma métrica derivada chamada índice de frustração, calculada como: (tempo de viagem × valor da tarifa) ÷ distância percorrida.

Essa fórmula simula a sensação de “pagar caro por pouco”, combinando três fatores que afetam diretamente a percepção do cliente. Agrupamos por zona de embarque (pickup_zip) e calculamos a média do índice para cada região.

Zonas com frustração média elevada indicam corridas longas, caras e pouco eficientes — o que pode gerar insatisfação mesmo quando a tarifa não é extrema. Esse indicador é um dos mais sensíveis da análise e compõe 30% da nota final simulada.

📌 Exemplo de achado: A zona 11430 apresenta frustração média de 219.65 — muito acima da média global — sugerindo um ponto crítico na jornada do cliente.

In [0]:
-- Índice de frustração médio por zona
SELECT 
  pickup_zip,
  ROUND(AVG((DATEDIFF(MINUTE, tpep_pickup_datetime, tpep_dropoff_datetime) * fare_amount) / trip_distance), 2) AS frustracao_media
FROM samples.nyctaxi.trips
WHERE pickup_zip IS NOT NULL AND trip_distance > 0
GROUP BY pickup_zip
ORDER BY frustracao_media DESC;


pickup_zip frustracao_media 11430 219.65 11424 177.03 11420 168.65 11218 136.36 7087 135.87 10305 135.0 11435 129.52 7737 122.09 11422 121.99 10111 120.18 10007 114.66 11423 114.29 10119 109.31 11221 108.69 10103 106.18 10065 104.32 11371 100.38 10154 98.97 11370 98.29 10199 97.65 11436 97.22 11375 95.49 7002 93.29 10014 92.23 11235 90.24 10017 86.73 11417 86.0 11109 83.8 10020 80.21 10001 79.89 10018 79.15 11226 79.01 10171 78.43 10011 74.44 11233 73.09 10025 72.83 11215 72.58 11377 71.6 10278 71.57 10009 71.22 10013 71.1 10022 70.85 11207 70.58 10112 70.56 10032 70.14 11368 70.11 10167 69.92 10271 68.24 10023 67.86 11101 67.7 10005 67.39 10012 67.37 10110 67.22 11222 66.74 10002 66.38 10006 65.42 11205 65.1 10282 64.84 10003 64.25 10152 64.09 11224 63.89 10038 63.88 10280 63.57 10021 63.39 10019 63.08 11225 62.59 10016 62.34 11216 61.01 10153 60.49 11369 60.2 10028 59.96 10470 59.83 11378 59.67 11104 59.0 10044 58.42 11231 58.2 11209 57.99 10115 57.65 11217 57.2 11373 56.56 10029 56.04 11211 55.38 11238 55.18 11230 55.17 10037 55.05 10030 54.95 11201 54.74 10026 54.35 10069 53.83 11102 51.91 10039 50.09 10162 49.33 10027 48.77 10034 48.64 11372 48.26 11103 47.69 11106 47.45 11237 47.32 10474 47.22 10452 47.03 11105 46.94 11206 46.39 10035 45.02 10031 44.8 11210 43.37 10451 40.78 10460 39.95 11355 38.37 11208 36.08 11356 33.33 11223 32.01 7718 25.0 10024 25.0 11418 19.48 11220 19.05 10454 17.65 11212 12.5 10468 11.22 7311 0.0 7030 0.0 10502 0.0 11003 0.0

### 📈 Proporção de corridas com tarifa acima da média
🔍 Pergunta
Quais zonas urbanas apresentam maior proporção de corridas com tarifa acima da média global?

🧠 Análise
Calculamos a média global das tarifas e, em seguida, verificamos em cada zona de embarque (pickup_zip) qual a proporção de corridas que ultrapassam esse valor. Essa métrica revela regiões onde o cliente tem maior chance de pagar mais do que o padrão da cidade.

Zonas com alta proporção de tarifas acima da média podem indicar práticas de precificação agressiva, rotas longas ou baixa eficiência. Esse indicador compõe 10% da nota final simulada e funciona como um alerta de recorrência — não basta uma corrida cara, mas sim o padrão repetido.

📌 Exemplo de achado: A zona 11368 apresenta 100% das corridas com tarifa acima da média — sugerindo um comportamento sistemático que merece atenção.

In [0]:
-- Proporção de corridas com tarifa acima da média
WITH media_global AS (
  SELECT AVG(fare_amount) AS media_tarifa FROM samples.nyctaxi.trips
)
SELECT 
  t.pickup_zip,
  ROUND(COUNT(CASE WHEN t.fare_amount > m.media_tarifa THEN 1 END) * 1.0 / COUNT(*), 2) AS proporcao_tarifa_alta
FROM samples.nyctaxi.trips t
JOIN media_global m ON 1=1
WHERE t.pickup_zip IS NOT NULL
GROUP BY t.pickup_zip
ORDER BY proporcao_tarifa_alta DESC;


pickup_zip proporcao_tarifa_alta 11368 1.00 8876 1.00 7310 1.00 7974 1.00 11356 1.00 10305 1.00 7114 1.00 11430 1.00 11420 1.00 11218 1.00 11417 1.00 7086 1.00 11207 1.00 10470 1.00 11208 1.00 7737 1.00 7030 1.00 11375 1.00 11221 1.00 11424 1.00 7311 1.00 7087 1.00 11422 0.97 11371 0.96 11436 0.82 11370 0.81 11373 0.75 10032 0.73 11215 0.67 10271 0.63 10005 0.63 10280 0.62 10006 0.61 10282 0.58 11222 0.57 11104 0.52 7002 0.50 11378 0.50 11235 0.50 11209 0.50 11226 0.50 10452 0.50 11210 0.50 11109 0.48 11102 0.47 10007 0.46 11101 0.45 10034 0.44 10278 0.44 11201 0.43 10038 0.43 11231 0.40 11225 0.40 10013 0.40 11211 0.39 11216 0.38 11377 0.38 10035 0.38 10002 0.36 11205 0.36 10115 0.35 11217 0.35 10171 0.34 11224 0.33 11233 0.33 11435 0.33 11237 0.33 10012 0.33 10103 0.33 10025 0.33 11372 0.33 11230 0.33 10037 0.31 11105 0.31 10003 0.31 10026 0.30 10154 0.30 10014 0.29 10044 0.29 10009 0.29 10018 0.29 10027 0.28 10030 0.28 10019 0.27 10020 0.27 10119 0.27 10029 0.27 10001 0.26 10028 0.26 10031 0.26 10167 0.26 11106 0.26 10016 0.26 10011 0.26 10021 0.25 10017 0.25 10110 0.25 11103 0.25 11238 0.25 10199 0.24 11206 0.24 10112 0.24 10111 0.24 10162 0.23 10022 0.22 10065 0.22 10023 0.21 11369 0.20 10039 0.20 10069 0.19 10153 0.18 10152 0.16 11003 0.00 11220 0.00 10502 0.00 11223 0.00 10460 0.00 11418 0.00 10024 0.00 10474 0.00 10463 0.00 11423 0.00 10451 0.00 11355 0.00 7718 0.00 11212 0.00 10454 0.00 10468 0.00

### 🧮 Etapas para calcular a nota final por zona
🔍 Pergunta

Como podemos consolidar os principais indicadores operacionais para avaliar a experiência do cliente por zona de embarque?

🧠 Análise

Nesta etapa, reunimos os quatro indicadores previamente calculados — tempo médio de viagem, tarifa média, índice de frustração e proporção de corridas caras — em uma única tabela por zona (pickup_zip). Cada métrica foi obtida por meio de uma subconsulta (CTE) e depois unida por meio de JOINs, garantindo que todas as zonas tenham seus respectivos valores consolidados.

Essa união é essencial para o próximo passo: a normalização e ponderação dos indicadores, que resultará na nota final simulada de experiência. Ao centralizar os dados em uma estrutura única, conseguimos aplicar regras de negócio e lógica de avaliação de forma consistente.

📌 Exemplo de achado: 

A zona 10305 apresenta tempo médio de 20 minutos, tarifa média de $13.50, frustração de 135.00 e 100% de corridas acima da média — um perfil crítico que será refletido na nota final.

In [0]:
-- Etapas para calcular a nota final
-- 1. Unir os quatro indicadores

WITH tempo AS (
  SELECT pickup_zip, 
         ROUND(AVG(DATEDIFF(MINUTE, tpep_pickup_datetime, tpep_dropoff_datetime)), 2) AS tempo_medio
  FROM samples.nyctaxi.trips
  WHERE pickup_zip IS NOT NULL
  GROUP BY pickup_zip
),
tarifa AS (
  SELECT pickup_zip, 
         ROUND(AVG(fare_amount), 2) AS tarifa_media
  FROM samples.nyctaxi.trips
  WHERE pickup_zip IS NOT NULL
  GROUP BY pickup_zip
),
frustracao AS (
  SELECT pickup_zip, 
         ROUND(AVG((DATEDIFF(MINUTE, tpep_pickup_datetime, tpep_dropoff_datetime) * fare_amount) / trip_distance), 2) AS frustracao_media
  FROM samples.nyctaxi.trips
  WHERE pickup_zip IS NOT NULL AND trip_distance > 0
  GROUP BY pickup_zip
),
corridas_caras AS (
  WITH media_global AS (
    SELECT AVG(fare_amount) AS media_tarifa FROM samples.nyctaxi.trips
  )
  SELECT t.pickup_zip,
         ROUND(COUNT(CASE WHEN t.fare_amount > m.media_tarifa THEN 1 END) * 1.0 / COUNT(*), 2) AS proporcao_tarifa_alta
  FROM samples.nyctaxi.trips t
  JOIN media_global m ON 1=1
  WHERE t.pickup_zip IS NOT NULL
  GROUP BY t.pickup_zip
)

SELECT 
  t.pickup_zip,
  t.tempo_medio,
  ta.tarifa_media,
  f.frustracao_media,
  c.proporcao_tarifa_alta
FROM tempo t
JOIN tarifa ta ON t.pickup_zip = ta.pickup_zip
JOIN frustracao f ON t.pickup_zip = f.pickup_zip
JOIN corridas_caras c ON t.pickup_zip = c.pickup_zip;


pickup_zip tempo_medio tarifa_media frustracao_media proporcao_tarifa_alta 11224 14.33 20.33 63.89 0.33 11368 12.0 52.0 70.11 1.00 10001 14.01 10.62 79.89 0.26 10021 11.02 10.21 63.39 0.25 11233 11.0 9.5 73.09 0.33 11231 11.2 12.03 58.2 0.40 10027 10.43 10.92 48.77 0.28 10028 11.8 10.21 59.96 0.26 11370 28.12 25.31 98.29 0.81 11003 0.0 0.0 0.0 0.00 7002 13.0 10.5 93.29 0.50 10023 11.88 10.04 67.86 0.21 10031 9.63 10.29 44.8 0.26 10271 15.94 15.8 68.24 0.63 10022 13.42 9.84 70.85 0.22 11378 7.0 27.25 59.67 0.50 10199 14.75 10.74 97.65 0.24 11220 4.0 6.0 19.05 0.00 10006 15.45 15.22 65.42 0.61 10037 12.96 13.4 55.05 0.31 11201 12.56 13.46 54.74 0.43 10502 0.0 0.0 0.0 0.00 11223 7.0 7.5 32.01 0.00 11356 10.0 15.0 33.33 1.00 10034 11.78 16.67 48.64 0.44 10305 20.0 13.5 135.0 1.00 10019 11.57 10.95 63.08 0.27 10002 14.36 12.23 66.38 0.36 10460 9.0 9.33 39.95 0.00 11206 10.52 11.02 46.39 0.24 11215 16.0 13.73 72.58 0.67 11418 3.0 5.0 19.48 0.00 10024 5.0 6.5 25.0 0.00 11430 73.25 52.0 219.65 1.00 10282 14.66 14.46 64.84 0.58 11235 29.5 33.25 90.24 0.50 10020 15.47 11.7 80.21 0.27 11373 15.5 16.63 56.56 0.75 10167 12.26 11.58 69.92 0.26 11420 48.0 52.0 168.65 1.00 11435 24.0 18.17 129.52 0.33 11218 30.0 20.0 136.36 1.00 11237 12.0 12.4 47.32 0.33 10007 22.72 13.86 114.66 0.46 11417 23.5 25.0 86.0 1.00 10474 10.0 8.5 47.22 0.00 11109 12.3 17.8 83.8 0.48 10038 15.15 15.9 63.88 0.43 10065 15.69 9.81 104.32 0.22 11207 19.5 20.5 70.58 1.00 10012 12.72 11.35 67.37 0.33 11209 15.5 19.25 57.99 0.50 10470 20.0 28.0 59.83 1.00 10115 10.93 11.89 57.65 0.35 11106 8.9 10.24 47.45 0.26 11216 14.88 15.63 61.01 0.38 11101 11.95 14.7 67.7 0.45 11225 15.6 14.6 62.59 0.40 11371 30.05 30.64 100.38 0.96 11423 16.0 11.5 114.29 0.00 11226 20.25 18.75 79.01 0.50 10119 18.43 10.91 109.31 0.27 10451 8.5 8.63 40.78 0.00 10016 11.83 10.96 62.34 0.26 10014 17.74 10.92 92.23 0.29 10044 11.44 10.91 58.42 0.29 11222 15.19 14.24 66.74 0.57 11355 9.0 11.0 38.37 0.00 10162 10.01 10.08 49.33 0.23 10017 13.53 11.31 86.73 0.25 11211 12.65 12.3 55.38 0.39 10032 16.27 16.63 70.14 0.73 7718 4.0 5.0 25.0 0.00 11212 2.0 4.0 12.5 0.00 10013 13.95 13.84 71.1 0.40 10452 12.0 13.88 47.03 0.50 10112 12.74 12.59 70.56 0.24 10103 18.5 12.61 106.18 0.33 11436 28.91 42.32 97.22 0.82 11208 12.0 19.0 36.08 1.00 11105 10.46 12.31 46.94 0.31 10069 10.4 9.78 53.83 0.19 10011 14.49 10.91 74.44 0.26 10152 11.28 10.13 64.09 0.16 10111 20.68 10.69 120.18 0.24 7737 23.0 15.5 122.09 1.00 10003 11.93 10.98 64.25 0.31 7030 0.0 40.0 0.0 1.00 10029 11.15 10.79 56.04 0.27 10280 15.33 16.33 63.57 0.62 10005 16.08 15.87 67.39 0.63 10110 11.73 10.9 67.22 0.25 11205 14.95 13.95 65.1 0.36 10026 10.94 10.4 54.35 0.30 11103 10.19 11.03 47.69 0.25 11238 11.94 11.44 55.18 0.25 10278 14.23 12.99 71.57 0.44 10025 16.88 11.68 72.83 0.33 10454 3.0 5.0 17.65 0.00 10153 10.81 9.77 60.49 0.18 11375 30.33 33.0 95.49 1.00 11369 7.8 24.2 60.2 0.20 10171 13.42 12.04 78.43 0.34 11377 15.29 13.25 71.6 0.38 10009 13.38 11.16 71.22 0.29 10035 9.95 13.27 45.02 0.38 10154 18.12 11.9 98.97 0.30 10030 10.28 10.75 54.95 0.28 10468 1.5 4.0 11.22 0.00 11104 13.07 12.24 59.0 0.52 11221 26.5 21.75 108.69 1.00 11422 41.47 44.78 121.99 0.97 11424 23.5 34.25 177.03 1.00 7311 0.0 60.0 0.0 1.00 11210 12.0 13.75 43.37 0.50 11372 10.53 12.37 48.26 0.33 11102 12.06 11.88 51.91 0.47 10039 7.6 7.7 50.09 0.20 11230 16.33 20.83 55.17 0.33 7087 38.0 31.0 135.87 1.00 10018 15.13 11.4 79.15 0.29 11217 12.3 12.25 57.2 0.35

### 📊 Normalização do tempo médio por zona

🔍 Pergunta

Como podemos transformar o tempo médio de viagem em uma escala comparável entre zonas, para facilitar a avaliação da experiência?

🧠 Análise

Nesta etapa, aplicamos uma técnica de normalização min-max ao tempo médio de viagem por zona (pickup_zip). A fórmula transforma os valores originais em uma escala de 0 a 10, onde 10 representa o melhor desempenho (menor tempo) e 0 o pior (maior tempo).

Essa abordagem permite comparar zonas com diferentes magnitudes de tempo de forma justa e padronizada. A normalização é essencial para compor a nota final, pois garante que todos os indicadores tenham o mesmo peso na escala de avaliação.

📌 Exemplo de achado:
 A zona 11220, com tempo médio de 4 minutos, recebeu nota 9.45 — indicando alta eficiência operacional.

In [0]:

-- 2. Normalizar os valores

WITH indicadores AS (
  SELECT 
    pickup_zip,
    ROUND(AVG(DATEDIFF(MINUTE, tpep_pickup_datetime, tpep_dropoff_datetime)), 2) AS tempo_medio
  FROM samples.nyctaxi.trips
  WHERE pickup_zip IS NOT NULL
  GROUP BY pickup_zip
),
limites AS (
  SELECT 
    MAX(tempo_medio) AS max_tempo,
    MIN(tempo_medio) AS min_tempo
  FROM indicadores
)
SELECT 
  i.pickup_zip,
  i.tempo_medio,
  ROUND(10 * (l.max_tempo - i.tempo_medio) / NULLIF((l.max_tempo - l.min_tempo), 0), 2) AS tempo_normalizado
FROM indicadores i
CROSS JOIN limites l
ORDER BY tempo_normalizado DESC;



pickup_zip tempo_medio tempo_normalizado 8876 0.0 10.0 11003 0.0 10.0 7114 0.0 10.0 7030 0.0 10.0 7086 0.0 10.0 10502 0.0 10.0 7311 0.0 10.0 7310 0.0 10.0 10463 0.0 10.0 7974 1.0 9.86 10468 1.5 9.8 11212 2.0 9.73 11418 3.0 9.59 10454 3.0 9.59 7718 4.0 9.45 11220 4.0 9.45 10024 5.0 9.32 11223 7.0 9.04 11378 7.0 9.04 10039 7.6 8.96 11369 7.8 8.94 10451 8.5 8.84 11106 8.9 8.78 10460 9.0 8.77 11355 9.0 8.77 10031 9.63 8.69 10035 9.95 8.64 11356 10.0 8.63 10474 10.0 8.63 10162 10.01 8.63 11103 10.19 8.61 10030 10.28 8.6 10027 10.43 8.58 10069 10.4 8.58 11105 10.46 8.57 11372 10.53 8.56 11206 10.52 8.56 10153 10.81 8.52 10026 10.94 8.51 10115 10.93 8.51 11233 11.0 8.5 10021 11.02 8.5 10029 11.15 8.48 11231 11.2 8.47 10152 11.28 8.46 10044 11.44 8.44 10019 11.57 8.42 10110 11.73 8.4 10034 11.78 8.39 10028 11.8 8.39 10023 11.88 8.38 10016 11.83 8.38 11101 11.95 8.37 11238 11.94 8.37 10003 11.93 8.37 11210 12.0 8.36 11237 12.0 8.36 11208 12.0 8.36 10452 12.0 8.36 11368 12.0 8.36 11102 12.06 8.35 10167 12.26 8.33 11109 12.3 8.32 11217 12.3 8.32 11201 12.56 8.29 11211 12.65 8.27 10012 12.72 8.26 10112 12.74 8.26 7002 13.0 8.23 10037 12.96 8.23 11104 13.07 8.22 10009 13.38 8.17 10171 13.42 8.17 10022 13.42 8.17 10017 13.53 8.15 10013 13.95 8.1 10001 14.01 8.09 10278 14.23 8.06 10002 14.36 8.04 11224 14.33 8.04 10011 14.49 8.02 10282 14.66 8.0 10199 14.75 7.99 11216 14.88 7.97 11205 14.95 7.96 10018 15.13 7.93 10038 15.15 7.93 11222 15.19 7.93 11377 15.29 7.91 10280 15.33 7.91 10020 15.47 7.89 10006 15.45 7.89 11209 15.5 7.88 11373 15.5 7.88 11225 15.6 7.87 10065 15.69 7.86 10271 15.94 7.82 11215 16.0 7.82 11423 16.0 7.82 10005 16.08 7.8 10032 16.27 7.78 11230 16.33 7.77 10025 16.88 7.7 10014 17.74 7.58 10154 18.12 7.53 10119 18.43 7.48 10103 18.5 7.47 11207 19.5 7.34 10470 20.0 7.27 10305 20.0 7.27 11226 20.25 7.24 10111 20.68 7.18 10007 22.72 6.9 7737 23.0 6.86 11424 23.5 6.79 11417 23.5 6.79 11435 24.0 6.72 11221 26.5 6.38 11370 28.12 6.16 11436 28.91 6.05 11235 29.5 5.97 11218 30.0 5.9 11371 30.05 5.9 11375 30.33 5.86 7087 38.0 4.81 11422 41.47 4.34 11420 48.0 3.45 11430 73.25 0.0

### 🧮 Cálculo da nota final por zona

🔍 Pergunta

Como podemos consolidar os indicadores operacionais em uma nota única que represente a experiência do cliente por zona?

🧠 Análise

Nesta etapa, criamos a nota final simulada para cada zona de embarque (pickup_zip), combinando quatro indicadores previamente calculados: tempo médio, tarifa média, índice de frustração e proporção de corridas caras. Cada indicador foi normalizado em uma escala de 0 a 10 e ponderado conforme seu impacto na experiência:

30% tempo médio

30% tarifa média

30% frustração média

10% proporção de tarifas altas

A fórmula utiliza a técnica de normalização min-max e aplica os pesos para gerar uma nota final entre 0 e 10, onde valores mais altos indicam melhor desempenho. Essa nota permite comparar zonas de forma justa e identificar áreas críticas com base em múltiplos fatores.

📌 Exemplo de achado: A zona 11106 recebeu nota 8.11, com frustração média de 47.45 e tarifa média de $11.07 — sugerindo um ponto de atenção na eficiência operacional.

In [0]:
-- 3. Calcular a nota final

WITH global_avg_fare AS (
  SELECT AVG(fare_amount) AS avg_fare
  FROM samples.nyctaxi.trips
),
base AS (
  SELECT 
    pickup_zip,
    ROUND(AVG(DATEDIFF(MINUTE, tpep_pickup_datetime, tpep_dropoff_datetime)), 2) AS tempo_medio,
    ROUND(AVG(fare_amount), 2) AS tarifa_media,
    ROUND(AVG((DATEDIFF(MINUTE, tpep_pickup_datetime, tpep_dropoff_datetime) * fare_amount) / trip_distance), 2) AS frustracao_media,
    ROUND(SUM(CASE WHEN fare_amount > g.avg_fare THEN 1 ELSE 0 END) * 1.0 / COUNT(*), 2) AS proporcao_tarifa_alta
  FROM samples.nyctaxi.trips
  CROSS JOIN global_avg_fare g
  WHERE pickup_zip IS NOT NULL AND trip_distance > 0
  GROUP BY pickup_zip
),
limites AS (
  SELECT 
    MAX(tempo_medio) AS max_tempo, MIN(tempo_medio) AS min_tempo,
    MAX(tarifa_media) AS max_tarifa, MIN(tarifa_media) AS min_tarifa,
    MAX(frustracao_media) AS max_frustracao, MIN(frustracao_media) AS min_frustracao,
    MAX(proporcao_tarifa_alta) AS max_proporcao, MIN(proporcao_tarifa_alta) AS min_proporcao
  FROM base
)
SELECT 
  b.pickup_zip,
  b.tempo_medio,
  b.tarifa_media,
  b.frustracao_media,
  b.proporcao_tarifa_alta,
  ROUND(
    0.3 * (10 * (l.max_tempo - b.tempo_medio) / NULLIF((l.max_tempo - l.min_tempo), 0)) +
    0.3 * (10 * (l.max_tarifa - b.tarifa_media) / NULLIF((l.max_tarifa - l.min_tarifa), 0)) +
    0.3 * (10 * (l.max_frustracao - b.frustracao_media) / NULLIF((l.max_frustracao - l.min_frustracao), 0)) +
    0.1 * (10 * (l.max_proporcao - b.proporcao_tarifa_alta) / NULLIF((l.max_proporcao - l.min_proporcao), 0))
  , 2) AS nota_final
FROM base b
CROSS JOIN limites l
ORDER BY nota_final DESC;

pickup_zip tempo_medio tarifa_media frustracao_media proporcao_tarifa_alta nota_final 11003 0.0 0.0 0.0 0.00 10.0 10502 0.0 0.0 0.0 0.00 10.0 10468 1.5 4.0 11.22 0.00 9.59 11212 2.0 4.0 12.5 0.00 9.55 10454 3.0 5.0 17.65 0.00 9.39 11418 3.0 5.0 19.48 0.00 9.36 11220 4.0 6.0 19.05 0.00 9.28 7718 4.0 5.0 25.0 0.00 9.24 10024 5.0 6.5 25.0 0.00 9.13 11223 7.0 7.5 32.01 0.00 8.9 10451 8.5 8.63 40.78 0.00 8.66 10460 9.0 9.33 39.95 0.00 8.62 11355 9.0 11.0 38.37 0.00 8.56 10474 10.0 8.5 47.22 0.00 8.52 10039 7.6 7.7 50.09 0.20 8.42 10031 9.63 10.29 44.8 0.26 8.22 10162 10.1 9.99 49.33 0.23 8.18 10069 10.4 9.78 53.83 0.19 8.16 11206 10.52 11.02 46.39 0.24 8.14 11103 10.19 11.03 47.69 0.25 8.13 11106 9.8 11.07 47.45 0.29 8.11 10027 10.54 11.03 48.77 0.29 8.06 10153 10.83 9.78 60.49 0.18 8.06 10030 10.28 10.75 54.95 0.28 8.01 10152 11.28 10.13 64.09 0.16 8.0 11105 10.46 12.31 46.94 0.31 8.0 10026 10.94 10.4 54.35 0.30 7.99 10035 10.21 12.25 45.02 0.37 7.98 10029 11.15 10.79 56.04 0.27 7.97 11372 10.53 12.37 48.26 0.33 7.96 11238 11.94 11.44 55.18 0.25 7.94 10028 11.81 10.16 59.96 0.26 7.93 10021 11.05 10.14 63.39 0.25 7.92 11237 12.0 12.4 47.32 0.33 7.91 10044 11.51 10.95 58.42 0.29 7.89 10023 11.9 10.05 67.86 0.21 7.87 10037 11.96 12.79 55.05 0.25 7.87 10016 11.83 10.96 62.34 0.26 7.86 10019 11.6 10.96 63.08 0.27 7.85 10115 10.93 11.89 57.65 0.35 7.82 10110 11.76 10.86 67.22 0.25 7.81 10022 13.43 9.85 70.85 0.22 7.77 10003 11.94 10.99 64.25 0.31 7.77 11231 11.43 11.87 58.2 0.39 7.75 11233 11.0 9.5 73.09 0.33 7.75 11217 12.5 12.4 57.2 0.35 7.74 11102 12.06 11.88 51.91 0.47 7.73 11210 12.0 13.75 43.37 0.50 7.73 11211 12.65 12.3 55.38 0.39 7.72 10167 12.3 11.61 69.92 0.26 7.7 10112 13.02 11.71 70.56 0.22 7.7 10452 12.0 13.88 47.03 0.50 7.67 10012 12.76 11.38 67.37 0.33 7.66 10009 13.44 11.13 71.22 0.29 7.63 11201 12.72 13.63 54.74 0.43 7.62 10034 11.78 16.67 48.64 0.44 7.58 10011 14.5 10.92 74.44 0.26 7.58 10001 14.01 10.62 79.89 0.26 7.54 10002 14.4 12.18 66.38 0.36 7.53 11104 13.07 12.24 59.0 0.52 7.53 10018 15.14 11.41 79.15 0.29 7.44 10171 13.42 12.04 78.43 0.34 7.44 11205 14.95 13.95 65.1 0.36 7.44 10017 13.57 11.33 86.73 0.25 7.44 10020 15.47 11.7 80.21 0.27 7.42 10025 16.88 11.68 72.83 0.33 7.4 11216 14.88 15.63 61.01 0.38 7.4 11356 10.0 15.0 33.33 1.00 7.39 11225 15.6 14.6 62.59 0.40 7.38 10013 13.95 13.84 71.1 0.40 7.37 11377 15.29 13.25 71.6 0.38 7.35 10278 14.23 12.99 71.57 0.44 7.35 10199 14.75 10.74 97.65 0.24 7.29 10038 15.41 16.18 63.88 0.44 7.25 11101 13.28 16.06 67.7 0.50 7.23 10065 15.71 9.75 104.32 0.22 7.22 11423 16.0 11.5 114.29 0.00 7.21 11230 16.33 20.83 55.17 0.33 7.21 10282 14.66 14.46 64.84 0.58 7.21 10014 17.74 10.92 92.23 0.29 7.18 11222 15.19 14.24 66.74 0.57 7.18 7002 13.0 10.5 93.29 0.50 7.17 11209 15.5 19.25 57.99 0.50 7.11 10006 15.57 15.32 65.42 0.61 7.09 10280 15.33 16.33 63.57 0.62 7.07 11208 12.0 19.0 36.08 1.00 7.07 11369 9.5 29.63 60.2 0.25 7.06 11378 7.0 27.25 59.67 0.50 7.04 10154 18.19 11.73 98.97 0.30 7.02 11373 15.5 16.63 56.56 0.75 7.01 11215 16.0 13.73 72.58 0.67 7.0 10271 16.06 15.52 68.24 0.63 7.0 10005 16.08 15.87 67.39 0.63 7.0 7030 0.0 40.0 0.0 1.00 7.0 11109 12.3 17.8 83.8 0.48 6.98 10119 18.46 10.92 109.31 0.27 6.93 10103 18.59 12.56 106.18 0.33 6.83 10032 16.27 16.63 70.14 0.73 6.81 10111 20.68 10.69 120.18 0.24 6.74 11226 20.25 18.75 79.01 0.50 6.65 10007 22.86 13.89 114.66 0.46 6.34 11224 21.5 29.25 63.89 0.50 6.28 11207 19.5 20.5 70.58 1.00 6.21 7311 0.0 60.0 0.0 1.00 6.0 10470 20.0 28.0 59.83 1.00 5.96 10305 20.0 13.5 135.0 1.00 5.66 7737 23.0 15.5 122.09 1.00 5.62 11417 23.5 25.0 86.0 1.00 5.61 11370 28.12 25.31 98.29 0.81 5.43 11235 29.5 33.25 90.24 0.50 5.4 11221 26.5 21.75 108.69 1.00 5.34 11435 36.0 26.0 129.52 0.50 4.96 11368 12.0 52.0 70.11 1.00 4.95 11218 30.0 20.0 136.36 1.00 4.91 11371 30.11 30.63 100.38 0.96 4.9 11375 30.33 33.0 95.49 1.00 4.8 11436 28.91 42.32 97.22 0.82 4.55 7087 38.0 31.0 135.87 1.00 4.04 11424 23.5 34.25 177.03 1.00 3.91 11422 42.13 45.23 121.99 0.98

### 🧾 Consolidação final: criação da view resultado_final

🔍 Pergunta

Como podemos reunir todos os indicadores normalizados em uma estrutura única que permita ranquear zonas urbanas pela experiência do cliente?

🧠 Análise

Nesta etapa, criamos a view temporária resultado_final, que consolida todos os indicadores operacionais por zona (pickup_zip) e calcula a nota final simulada. A nota é composta por quatro métricas:

Tempo médio de viagem

Tarifa média

Índice de frustração

Proporção de corridas caras

Cada métrica é normalizada em uma escala de 0 a 10 usando a técnica min-max, e ponderada conforme seu impacto na experiência:

30% tempo

30% tarifa

30% frustração

10% proporção de tarifas altas

A view permite consultar diretamente as zonas com melhor ou pior desempenho, facilitando a criação de rankings, alertas e visualizações. É o ponto de partida para a aba “Zonas em Alerta” do dashboard corporativo.

📌 Exemplo de achado: A zona 11106 recebeu nota 8.11, com frustração média de 47.45 e tarifa média de $11.07 — sugerindo um ponto de atenção na eficiência operacional.

In [0]:
CREATE OR REPLACE TEMP VIEW resultado_final AS
WITH global_avg_fare AS (
  SELECT AVG(fare_amount) AS avg_fare FROM samples.nyctaxi.trips
),
base AS (
  SELECT 
    pickup_zip,
    ROUND(AVG(DATEDIFF(MINUTE, tpep_pickup_datetime, tpep_dropoff_datetime)), 2) AS tempo_medio,
    ROUND(AVG(fare_amount), 2) AS tarifa_media,
    ROUND(AVG((DATEDIFF(MINUTE, tpep_pickup_datetime, tpep_dropoff_datetime) * fare_amount) / trip_distance), 2) AS frustracao_media,
    ROUND(SUM(CASE WHEN fare_amount > g.avg_fare THEN 1 ELSE 0 END) * 1.0 / COUNT(*), 2) AS proporcao_tarifa_alta
  FROM samples.nyctaxi.trips
  CROSS JOIN global_avg_fare g
  WHERE pickup_zip IS NOT NULL AND trip_distance > 0
  GROUP BY pickup_zip
),
limites AS (
  SELECT 
    MAX(tempo_medio) AS max_tempo, MIN(tempo_medio) AS min_tempo,
    MAX(tarifa_media) AS max_tarifa, MIN(tarifa_media) AS min_tarifa,
    MAX(frustracao_media) AS max_frustracao, MIN(frustracao_media) AS min_frustracao,
    MAX(proporcao_tarifa_alta) AS max_proporcao, MIN(proporcao_tarifa_alta) AS min_proporcao
  FROM base
)
SELECT 
  b.pickup_zip,
  b.tempo_medio,
  b.tarifa_media,
  b.frustracao_media,
  b.proporcao_tarifa_alta,
  ROUND(
    0.3 * (10 * (l.max_tempo - b.tempo_medio) / NULLIF((l.max_tempo - l.min_tempo), 0)) +
    0.3 * (10 * (l.max_tarifa - b.tarifa_media) / NULLIF((l.max_tarifa - l.min_tarifa), 0)) +
    0.3 * (10 * (l.max_frustracao - b.frustracao_media) / NULLIF((l.max_frustracao - l.min_frustracao), 0)) +
    0.1 * (10 * (l.max_proporcao - b.proporcao_tarifa_alta) / NULLIF((l.max_proporcao - l.min_proporcao), 0))
  , 2) AS nota_final
FROM base b
CROSS JOIN limites l;


### 🏁 Consulta final: ranking de zonas por nota de experiência

🔍 Pergunta

Quais zonas urbanas apresentam melhor desempenho na experiência do cliente, considerando todos os indicadores combinados?

🧠 Análise

Este comando consulta a view resultado_final e ordena as zonas de embarque (pickup_zip) pela nota final simulada, em ordem decrescente. A nota foi calculada com base em quatro indicadores normalizados e ponderados: tempo médio, tarifa média, índice de frustração e proporção de corridas caras.

Essa consulta permite identificar rapidamente as zonas com melhor e pior desempenho, funcionando como um ranking operacional. É o ponto de partida para visualizações estratégicas, alertas e decisões gerenciais.

📌 Exemplo de achado: A zona 11003 recebeu nota 10, com todos os indicadores zerados — sugerindo um caso atípico ou ausência de dados. Já a zona 11106 aparece com nota 8.11, indicando desempenho abaixo do ideal e merecendo atenção.

In [0]:
SELECT * FROM resultado_final ORDER BY nota_final DESC


pickup_zip tempo_medio tarifa_media frustracao_media proporcao_tarifa_alta nota_final 11003 0.0 0.0 0.0 0.00 10.0 10502 0.0 0.0 0.0 0.00 10.0 10468 1.5 4.0 11.22 0.00 9.59 11212 2.0 4.0 12.5 0.00 9.55 10454 3.0 5.0 17.65 0.00 9.39 11418 3.0 5.0 19.48 0.00 9.36 11220 4.0 6.0 19.05 0.00 9.28 7718 4.0 5.0 25.0 0.00 9.24 10024 5.0 6.5 25.0 0.00 9.13 11223 7.0 7.5 32.01 0.00 8.9 10451 8.5 8.63 40.78 0.00 8.66 10460 9.0 9.33 39.95 0.00 8.62 11355 9.0 11.0 38.37 0.00 8.56 10474 10.0 8.5 47.22 0.00 8.52 10039 7.6 7.7 50.09 0.20 8.42 10031 9.63 10.29 44.8 0.26 8.22 10162 10.1 9.99 49.33 0.23 8.18 10069 10.4 9.78 53.83 0.19 8.16 11206 10.52 11.02 46.39 0.24 8.14 11103 10.19 11.03 47.69 0.25 8.13 11106 9.8 11.07 47.45 0.29 8.11 10027 10.54 11.03 48.77 0.29 8.06 10153 10.83 9.78 60.49 0.18 8.06 10030 10.28 10.75 54.95 0.28 8.01 10152 11.28 10.13 64.09 0.16 8.0 11105 10.46 12.31 46.94 0.31 8.0 10026 10.94 10.4 54.35 0.30 7.99 10035 10.21 12.25 45.02 0.37 7.98 10029 11.15 10.79 56.04 0.27 7.97 11372 10.53 12.37 48.26 0.33 7.96 11238 11.94 11.44 55.18 0.25 7.94 10028 11.81 10.16 59.96 0.26 7.93 10021 11.05 10.14 63.39 0.25 7.92 11237 12.0 12.4 47.32 0.33 7.91 10044 11.51 10.95 58.42 0.29 7.89 10023 11.9 10.05 67.86 0.21 7.87 10037 11.96 12.79 55.05 0.25 7.87 10016 11.83 10.96 62.34 0.26 7.86 10019 11.6 10.96 63.08 0.27 7.85 10115 10.93 11.89 57.65 0.35 7.82 10110 11.76 10.86 67.22 0.25 7.81 10022 13.43 9.85 70.85 0.22 7.77 10003 11.94 10.99 64.25 0.31 7.77 11231 11.43 11.87 58.2 0.39 7.75 11233 11.0 9.5 73.09 0.33 7.75 11217 12.5 12.4 57.2 0.35 7.74 11102 12.06 11.88 51.91 0.47 7.73 11210 12.0 13.75 43.37 0.50 7.73 11211 12.65 12.3 55.38 0.39 7.72 10167 12.3 11.61 69.92 0.26 7.7 10112 13.02 11.71 70.56 0.22 7.7 10452 12.0 13.88 47.03 0.50 7.67 10012 12.76 11.38 67.37 0.33 7.66 10009 13.44 11.13 71.22 0.29 7.63 11201 12.72 13.63 54.74 0.43 7.62 10034 11.78 16.67 48.64 0.44 7.58 10011 14.5 10.92 74.44 0.26 7.58 10001 14.01 10.62 79.89 0.26 7.54 10002 14.4 12.18 66.38 0.36 7.53 11104 13.07 12.24 59.0 0.52 7.53 10018 15.14 11.41 79.15 0.29 7.44 10171 13.42 12.04 78.43 0.34 7.44 11205 14.95 13.95 65.1 0.36 7.44 10017 13.57 11.33 86.73 0.25 7.44 10020 15.47 11.7 80.21 0.27 7.42 10025 16.88 11.68 72.83 0.33 7.4 11216 14.88 15.63 61.01 0.38 7.4 11356 10.0 15.0 33.33 1.00 7.39 11225 15.6 14.6 62.59 0.40 7.38 10013 13.95 13.84 71.1 0.40 7.37 11377 15.29 13.25 71.6 0.38 7.35 10278 14.23 12.99 71.57 0.44 7.35 10199 14.75 10.74 97.65 0.24 7.29 10038 15.41 16.18 63.88 0.44 7.25 11101 13.28 16.06 67.7 0.50 7.23 10065 15.71 9.75 104.32 0.22 7.22 11423 16.0 11.5 114.29 0.00 7.21 11230 16.33 20.83 55.17 0.33 7.21 10282 14.66 14.46 64.84 0.58 7.21 10014 17.74 10.92 92.23 0.29 7.18 11222 15.19 14.24 66.74 0.57 7.18 7002 13.0 10.5 93.29 0.50 7.17 11209 15.5 19.25 57.99 0.50 7.11 10006 15.57 15.32 65.42 0.61 7.09 10280 15.33 16.33 63.57 0.62 7.07 11208 12.0 19.0 36.08 1.00 7.07 11369 9.5 29.63 60.2 0.25 7.06 11378 7.0 27.25 59.67 0.50 7.04 10154 18.19 11.73 98.97 0.30 7.02 11373 15.5 16.63 56.56 0.75 7.01 11215 16.0 13.73 72.58 0.67 7.0 10271 16.06 15.52 68.24 0.63 7.0 10005 16.08 15.87 67.39 0.63 7.0 7030 0.0 40.0 0.0 1.00 7.0 11109 12.3 17.8 83.8 0.48 6.98 10119 18.46 10.92 109.31 0.27 6.93 10103 18.59 12.56 106.18 0.33 6.83 10032 16.27 16.63 70.14 0.73 6.81 10111 20.68 10.69 120.18 0.24 6.74 11226 20.25 18.75 79.01 0.50 6.65 10007 22.86 13.89 114.66 0.46 6.34 11224 21.5 29.25 63.89 0.50 6.28 11207 19.5 20.5 70.58 1.00 6.21 7311 0.0 60.0 0.0 1.00 6.0 10470 20.0 28.0 59.83 1.00 5.96 10305 20.0 13.5 135.0 1.00 5.66 7737 23.0 15.5 122.09 1.00 5.62 11417 23.5 25.0 86.0 1.00 5.61 11370 28.12 25.31 98.29 0.81 5.43 11235 29.5 33.25 90.24 0.50 5.4 11221 26.5 21.75 108.69 1.00 5.34 11435 36.0 26.0 129.52 0.50 4.96 11368 12.0 52.0 70.11 1.00 4.95 11218 30.0 20.0 136.36 1.00 4.91 11371 30.11 30.63 100.38 0.96 4.9 11375 30.33 33.0 95.49 1.00 4.8 11436 28.91 42.32 97.22 0.82 4.55 7087 38.0 31.0 135.87 1.00 4.04 11424 23.5 34.25 177.03 1.00 3.91 11422 42.13 45.23 121.99 0.98

Databricks visualization. Run in Databricks to view.

Databricks visualization. Run in Databricks to view.

### Zonas em Alerta

Esta aba apresenta as zonas urbanas com **nota final abaixo de 6**, indicando risco elevado de insatisfação na jornada do cliente. A nota foi calculada com base em quatro indicadores operacionais:

- Tempo médio de viagem  
- Tarifa média  
- Índice de frustração  
- Proporção de corridas caras

Cada métrica foi normalizada e ponderada conforme seu impacto na experiência. Zonas com nota baixa tendem a apresentar ineficiências operacionais, como corridas longas, caras ou pouco eficientes.

#### 🎯 Objetivo
Permitir que líderes operacionais identifiquem rapidamente os pontos críticos e tomem decisões baseadas em dados.

#### 📌 Recomendação
Monitorar zonas com nota < 6, revisar rotas e precificação, e investigar causas sistêmicas que afetam a experiência.

> Clique em uma zona para visualizar os detalhes operacionais e histórico de desempenho.


### Tabela de Zonas Críticas

Esta tabela apresenta os indicadores operacionais por zona de embarque. As zonas com nota final abaixo de 6 são consideradas críticas e estão destacadas em vermelho.

A tabela permite comparar os fatores que mais impactam a experiência do cliente em cada região — seja tempo de viagem, tarifa, frustração ou recorrência de corridas caras.

> Use os filtros para explorar zonas específicas ou ajustar o limite de alerta.


### Ranking de Zonas por Nota Final

Este gráfico apresenta as zonas urbanas ordenadas pela nota final simulada de experiência. A nota foi calculada com base em quatro indicadores operacionais: tempo médio, tarifa média, índice de frustração e proporção de corridas caras.

Zonas com nota abaixo de 6 são consideradas críticas e devem ser monitoradas com atenção. A visualização permite identificar rapidamente os pontos de maior risco na jornada do cliente.

> Clique em uma zona para visualizar os detalhes operacionais.


### Experiência Geral nas Zonas Urbanas

Este gráfico de pizza simula um indicador circular da média de experiência nas zonas urbanas. A nota é composta por quatro métricas operacionais e varia de 0 a 10.

A fatia verde representa a média atual de desempenho. A fatia cinza mostra o quanto falta para atingir a nota máxima.

> Use este indicador como termômetro geral da operação.


In [0]:
SELECT 'Experiência' AS categoria, ROUND(AVG(nota_final), 2) AS valor
FROM resultado_final
UNION ALL
SELECT 'Gap', ROUND(10 - AVG(nota_final), 2) AS valor
FROM resultado_final


categoria valor Experiência 7.24 Gap 2.76

Databricks visualization. Run in Databricks to view.

In [0]:
CREATE OR REPLACE TABLE resultado_final_export AS
SELECT * FROM resultado_final


num_affected_rows num_inserted_rows

### 🔄 Exportar dados para Power BI

Execute a célula abaixo e clique em “Download CSV” para obter o arquivo completo com todas as zonas urbanas e seus indicadores. Use este arquivo como base para montar os visuais no Power BI.



In [0]:
-- Exportação completa para Power BI
SELECT * FROM resultado_final_export

pickup_zip tempo_medio tarifa_media frustracao_media proporcao_tarifa_alta nota_final 11224 21.5 29.25 63.89 0.50 6.28 11368 12.0 52.0 70.11 1.00 4.95 10001 14.01 10.62 79.89 0.26 7.54 10021 11.05 10.14 63.39 0.25 7.92 11233 11.0 9.5 73.09 0.33 7.75 11231 11.43 11.87 58.2 0.39 7.75 10027 10.54 11.03 48.77 0.29 8.06 10028 11.81 10.16 59.96 0.26 7.93 11370 28.12 25.31 98.29 0.81 5.43 11003 0.0 0.0 0.0 0.00 10.0 7002 13.0 10.5 93.29 0.50 7.17 10023 11.9 10.05 67.86 0.21 7.87 10031 9.63 10.29 44.8 0.26 8.22 10271 16.06 15.52 68.24 0.63 7.0 10022 13.43 9.85 70.85 0.22 7.77 11378 7.0 27.25 59.67 0.50 7.04 10199 14.75 10.74 97.65 0.24 7.29 11220 4.0 6.0 19.05 0.00 9.28 10006 15.57 15.32 65.42 0.61 7.09 10037 11.96 12.79 55.05 0.25 7.87 11201 12.72 13.63 54.74 0.43 7.62 10502 0.0 0.0 0.0 0.00 10.0 11223 7.0 7.5 32.01 0.00 8.9 11356 10.0 15.0 33.33 1.00 7.39 10034 11.78 16.67 48.64 0.44 7.58 10305 20.0 13.5 135.0 1.00 5.66 10019 11.6 10.96 63.08 0.27 7.85 10002 14.4 12.18 66.38 0.36 7.53 10460 9.0 9.33 39.95 0.00 8.62 11206 10.52 11.02 46.39 0.24 8.14 11215 16.0 13.73 72.58 0.67 7.0 11418 3.0 5.0 19.48 0.00 9.36 10024 5.0 6.5 25.0 0.00 9.13 11430 73.25 52.0 219.65 1.00 0.4 10282 14.66 14.46 64.84 0.58 7.21 11235 29.5 33.25 90.24 0.50 5.4 10020 15.47 11.7 80.21 0.27 7.42 11373 15.5 16.63 56.56 0.75 7.01 10167 12.3 11.61 69.92 0.26 7.7 11420 48.0 52.0 168.65 1.00 2.13 11435 36.0 26.0 129.52 0.50 4.96 11218 30.0 20.0 136.36 1.00 4.91 11237 12.0 12.4 47.32 0.33 7.91 10007 22.86 13.89 114.66 0.46 6.34 11417 23.5 25.0 86.0 1.00 5.61 10474 10.0 8.5 47.22 0.00 8.52 11109 12.3 17.8 83.8 0.48 6.98 10038 15.41 16.18 63.88 0.44 7.25 10065 15.71 9.75 104.32 0.22 7.22 11207 19.5 20.5 70.58 1.00 6.21 10012 12.76 11.38 67.37 0.33 7.66 11209 15.5 19.25 57.99 0.50 7.11 10470 20.0 28.0 59.83 1.00 5.96 10115 10.93 11.89 57.65 0.35 7.82 11106 9.8 11.07 47.45 0.29 8.11 11216 14.88 15.63 61.01 0.38 7.4 11101 13.28 16.06 67.7 0.50 7.23 11225 15.6 14.6 62.59 0.40 7.38 11371 30.11 30.63 100.38 0.96 4.9 11423 16.0 11.5 114.29 0.00 7.21 11226 20.25 18.75 79.01 0.50 6.65 10119 18.46 10.92 109.31 0.27 6.93 10451 8.5 8.63 40.78 0.00 8.66 10016 11.83 10.96 62.34 0.26 7.86 10014 17.74 10.92 92.23 0.29 7.18 10044 11.51 10.95 58.42 0.29 7.89 11222 15.19 14.24 66.74 0.57 7.18 11355 9.0 11.0 38.37 0.00 8.56 10162 10.1 9.99 49.33 0.23 8.18 10017 13.57 11.33 86.73 0.25 7.44 11211 12.65 12.3 55.38 0.39 7.72 10032 16.27 16.63 70.14 0.73 6.81 7718 4.0 5.0 25.0 0.00 9.24 11212 2.0 4.0 12.5 0.00 9.55 10013 13.95 13.84 71.1 0.40 7.37 10452 12.0 13.88 47.03 0.50 7.67 10112 13.02 11.71 70.56 0.22 7.7 10103 18.59 12.56 106.18 0.33 6.83 11436 28.91 42.32 97.22 0.82 4.55 11208 12.0 19.0 36.08 1.00 7.07 11105 10.46 12.31 46.94 0.31 8.0 10069 10.4 9.78 53.83 0.19 8.16 10011 14.5 10.92 74.44 0.26 7.58 10152 11.28 10.13 64.09 0.16 8.0 10111 20.68 10.69 120.18 0.24 6.74 7737 23.0 15.5 122.09 1.00 5.62 10003 11.94 10.99 64.25 0.31 7.77 7030 0.0 40.0 0.0 1.00 7.0 10029 11.15 10.79 56.04 0.27 7.97 10280 15.33 16.33 63.57 0.62 7.07 10005 16.08 15.87 67.39 0.63 7.0 10110 11.76 10.86 67.22 0.25 7.81 11205 14.95 13.95 65.1 0.36 7.44 10026 10.94 10.4 54.35 0.30 7.99 11103 10.19 11.03 47.69 0.25 8.13 11238 11.94 11.44 55.18 0.25 7.94 10278 14.23 12.99 71.57 0.44 7.35 10025 16.88 11.68 72.83 0.33 7.4 10454 3.0 5.0 17.65 0.00 9.39 10153 10.83 9.78 60.49 0.18 8.06 11375 30.33 33.0 95.49 1.00 4.8 11369 9.5 29.63 60.2 0.25 7.06 10171 13.42 12.04 78.43 0.34 7.44 11377 15.29 13.25 71.6 0.38 7.35 10009 13.44 11.13 71.22 0.29 7.63 10035 10.21 12.25 45.02 0.37 7.98 10154 18.19 11.73 98.97 0.30 7.02 10030 10.28 10.75 54.95 0.28 8.01 10468 1.5 4.0 11.22 0.00 9.59 11104 13.07 12.24 59.0 0.52 7.53 11221 26.5 21.75 108.69 1.00 5.34 11422 42.13 45.23 121.99 0.98 3.37 11424 23.5 34.25 177.03 1.00 3.91 7311 0.0 60.0 0.0 1.00 6.0 11210 12.0 13.75 43.37 0.50 7.73 11372 10.53 12.37 48.26 0.33 7.96 11102 12.06 11.88 51.91 0.47 7.73 10039 7.6 7.7 50.09 0.20 8.42 11230 16.33 20.83 55.17 0.33 7.21 7087 38.0 31.0 135.87 1.0